From: Felix Krull <f_krull@gmx.de>
Date: Fri, 22 Dec 2017 08:45:08 -0800
Subject: deadsnakes-no-rpath

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7ae0f53..fd7eff0 100644
--- a/setup.py
+++ b/setup.py
@@ -1193,7 +1193,7 @@ class PyBuildExt(build_ext):
                         if dbm_setup_debug: print("building dbm using bdb")
                         dbmext = Extension('_dbm', ['_dbmmodule.c'],
                                            library_dirs=dblib_dir,
-                                           runtime_library_dirs=dblib_dir,
+                                           #runtime_library_dirs=dblib_dir,
                                            include_dirs=db_incs,
                                            define_macros=[
                                                ('HAVE_BERKDB_H', None),
