# DP: Mention the FAQ on the documentation index page.

--- a/Doc/html/index.html.in~	2002-04-01 18:11:27.000000000 +0200
+++ b/Doc/html/index.html.in	2003-04-05 13:33:35.000000000 +0200
@@ -123,6 +123,24 @@
           </ul>
           </td>
         </tr>
+        <tr>
+          <td valign="baseline" class="left-column">
+          &nbsp;
+          <ul>
+            <li> <a href="../../python/FAQ.html" class="title"
+                >FAQ (local copy)</a>
+              <br>(python package must be installed)
+          </ul>
+          </td>
+          <td valign="baseline" class="right-column">
+          &nbsp;
+          <ul>
+            <li> <a href="http://www.python.org/cgi-bin/faqw.py" class="title"
+                >FAQ (online wizard)</a>
+              <br>(maybe more recent)
+          </ul>
+          </td>
+        </tr>
       </tbody>
     </table>
     <p>
