From: Felix Krull <f_krull@gmx.de>
Date: Fri, 22 Dec 2017 08:45:07 -0800
Subject: hurd-path_max

---
 Python/pythonrun.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Python/pythonrun.c b/Python/pythonrun.c
index 91d56b7..49bbb2a 100644
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -35,6 +35,10 @@
 #define PATH_MAX MAXPATHLEN
 #endif
 
+#ifdef __gnu_hurd__
+#define PATH_MAX MAXPATHLEN
+#endif
+
 #ifndef Py_REF_DEBUG
 #define PRINT_TOTAL_REFS()
 #else /* Py_REF_DEBUG */
