# DP: argparse.py: Make the gettext import conditional

--- a/Lib/argparse.py
+++ b/Lib/argparse.py
@@ -90,7 +90,14 @@
 import sys as _sys
 import textwrap as _textwrap
 
-from gettext import gettext as _, ngettext
+try:
+    from gettext import gettext, ngettext
+except ImportError:
+    def gettext(message):
+        return message
+    def ngettext(msg1, msg2, n):
+        return msg1 if n == 1 else msg2
+_ = gettext
 
 
 SUPPRESS = '==SUPPRESS=='
