From: Felix Krull <f_krull@gmx.de>
Date: Thu, 21 Dec 2017 18:40:41 -0800
Subject: libffi-shared

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 882ee1e..c3eaa29 100644
--- a/setup.py
+++ b/setup.py
@@ -1932,7 +1932,7 @@ class PyBuildExt(build_ext):
                         break
         ffi_lib = None
         if ffi_inc is not None:
-            for lib_name in ('ffi_convenience', 'ffi_pic', 'ffi'):
+            for lib_name in ('ffi', 'ffi_convenience', 'ffi_pic', 'ffi'):
                 if (self.compiler.find_library_file(lib_dirs, lib_name)):
                     ffi_lib = lib_name
                     break
